### DISCUSSION TITLE
Asking about "reference" field in Inventory Detail

### DISCUSSION THREAD

**Discussion by muhridwan-kk** [frePPLe/frepple#666] [General]
Hi All,
I understand that there are 4 valid values for InventoryDetail->reference field: MO/PO/DO/DLVR
My question is: how to put value for stock in hand?
those 4 values, in my understanding, refer to 4 activities (Manufacturing Order, Purchase Order, Distribution Order and Delivery).
Now when I setup for the first time, what should I put the value for stock in hand?
ref: https://frepple.com/docs/current/model-reference/operationplan-materials.php

thank you

**Comment by jdetaeye** [comment]
Your current inventory is fed into in the buffer.onhand field.
That will automatically generate a record of type STCK.

In the Enterprise & Cloud Editions there is an extra app "shelf life".  It allows you to provide your inventory by "lots" or "batch", each with an expiry date:  eg "reference 1234 of 50 pieces was put on stock on 2024-07-15 and will expire on 2024-08-15".    A separate table is used to store these.

**Comment by muhridwan-kk** [comment]
Thanks for the answer.
I am doing trial, so it is an Enterprise version.

> Your current inventory is fed into in the buffer.onhand field.
> That will automatically generate a record of type STCK.

Ok, I'll try this first
